jest.config.ts

```diff
@@ -37,16 +37,84 @@ function jestGithubRunnerSpecs(): JestConfig {
   };
 }
 
+interface ShardConfig {
+  patterns: string[];
+  threshold?: {
+    branches?: number;
+    functions?: number;
+    lines?: number;
+    statements?: number;
+  };
+}
+
+type JestShardedSubconfig = Pick<
+  JestConfig,
+  'testMatch' | 'collectCoverageFrom' | 'coverageThreshold'
+>;
+
+const shardMap: Record<string, ShardConfig> = {
+  datasource: { patterns: ['lib/modules/datasource'] },
+  manager_1: { patterns: ['lib/modules/manager/[a-m]*'] },
+  manager_2: { patterns: ['lib/modules/manager/[n-z]*'] },
+  platform: { patterns: ['lib/modules/platform'] },
+  versioning: { patterns: ['lib/modules/versioning'] },
+  workers: { patterns: ['lib/workers'] },
+  other: { patterns: ['lib'] },
+};
```

Comment:
### Are we sure that all sharded tests have run?

It looks like you're manually setting a `shardMap`? I have some questions:

- How do we know that we have ran all the tests at the end?
- What happens if a directory which should be tested is _not_ on that `shardMap`?

Reply:
Each shard item includes everything that matches its pattern, except everything from previously defined shards. It means, `other` should test everything that wasn't defined earlier.